### PR TITLE
chore: add junit5 extension to setup environment variable

### PR DESCRIPTION
**Pull request by JoeWang1127**
Fix: #2785.

### CODE REVIEW

File: gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java [hunk 1]
```diff
@@ -339,6 +340,7 @@ void endpointContextBuild_gdchFlow_noUniverseDomain_customEndpoint() throws IOEx
   // For this test running locally or in CI, check that the Env Var is set properly.
   // This test should only run when the maven profile `EnvVarTest` is enabled.
```
Comment (lqiu96) [inline gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java:341]: Can you remove this line: `// This test should only run when the maven profile `EnvVarTest` is enabled.`

File: gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java [hunk 2]
```diff
@@ -354,9 +356,9 @@ void endpointContextBuild_universeDomainEnvVarSet() throws IOException {
...
```
Comment (lqiu96) [inline gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java:357]: Same here. Remove this line.

File: gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java [hunk 3]
```diff
@@ -339,6 +340,7 @@ void endpointContextBuild_gdchFlow_noUniverseDomain_customEndpoint() throws IOEx
...
   @Test
+  @SetEnvironmentVariable(key = "GOOGLE_CLOUD_UNIVERSE_DOMAIN", value = "random.com")
```
Comment (lqiu96) [inline gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java:343]: If possible, can the key be something like `key = EndpointContext.GOOGLE_CLOUD_UNIVERSE_DOMAIN` using the constant from https://github.com/googleapis/sdk-platform-java/blob/cf44b0754258b18fe3ae29725e9713b548beff80/gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java#L62

**Review by lqiu96 — APPROVED** [review]
LGTM

File: gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java (hunk 3 above)
Comment (JoeWang1127) [reply gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java]: Done.

File: gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java (hunk 2 above)
Comment (JoeWang1127) [reply gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java]: Done.

File: gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java (hunk 1 above)
Comment (JoeWang1127) [reply gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java]: Done.

**Pull request merged**